### PR TITLE
Refactor Emojis by removing redundant codes

### PR DESCRIPTION
**Pull request by pankona**
Refs: https://github.com/mobu-of-the-world/mobu/pull/479

## Updates

Refactor to remove redundant codes related to Emojis

### CODE REVIEW

File: src/common/Emoji.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ const Emoji = ({
   emojiName,
   onClick,
 }: {
-  emojiName: EmojiName;
+  emojiName: keyof typeof EmojiName;
```
Comment (pankona) [inline src/common/Emoji.tsx:10]: 📝 only accepts "Car", "Speaker", "Wastebasket", "Plus" or "CrossMark"

**Comment by pankona** [comment]
@kachick 

Please review and merge if there's no concern! 😋

File: src/userList/User.tsx [hunk 2]
```diff
@@ -13,11 +13,11 @@ const User = ({ isDriver, user }: { isDriver: boolean; user: string }) => {
   );
 };
 
-const emojiNameByRole = (isDriver: boolean): EmojiName => {
+const emojiNameByRole = (isDriver: boolean) => {
```
Comment (kachick) [inline src/userList/User.tsx:16]: Why removing type annotation from here? Keeping it makes clear.

File: src/common/Emoji.tsx [hunk 3]
```diff
@@ -18,43 +18,17 @@ const Emoji = ({
...
-  Speaker: "Speaker",
-  Wastebasket: "Wastebasket",
-  Plus: "Plus",
-  CrossMark: "CrossMark",
+  Car: "🚗",
+  Speaker: "🗣",
+  Wastebasket: "🗑",
+  Plus: "➕",
+  CrossMark: "❌",
 } as const;
```
Comment (kachick) [inline src/common/Emoji.tsx:32]: Good refactoring 👍

File: src/common/Emoji.tsx [hunk 4]
```diff
@@ -18,43 +18,17 @@ const Emoji = ({
       ])}
       onClick={onClick}
     >
-      {emojiByName(emojiName)}
+      {EmojiName[emojiName]}
     </div>
   );
 };
 
 export const EmojiName = {
```
Comment (kachick) [inline src/common/Emoji.tsx:26]:
```suggestion
export const EmojiByName = {
```

How about this for new object?

File: src/userList/User.tsx (hunk 2 above)
Comment (kachick) [reply src/userList/User.tsx:16]: Or removing this function makes simple.

File: src/common/Emoji.tsx (hunk 4 above)
Comment (kachick) [reply src/common/Emoji.tsx:26]: Still need the exporting?

File: src/userList/User.tsx (hunk 2 above)
Comment (pankona) [reply src/userList/User.tsx]: Make sense. I replaced this function with ternary operator. 
Addressed in https://github.com/mobu-of-the-world/mobu/pull/481/commits/63b264d36ab6e62aad02ebf0f0d495816010ce8b

File: src/common/Emoji.tsx (hunk 4 above)
Comment (pankona) [reply src/common/Emoji.tsx]: Thanks. Addressed in https://github.com/mobu-of-the-world/mobu/pull/481/commits/abe0df5e49ab8538f5730b1f394b2b8ba24554bc

**Comment by pankona** [comment]
@kachick 

Thank you for your review! I addressed all your points.
Could you check again?

**Review by kachick — APPROVED** [review]
LGTM 👍 Thank you!

**Pull request merged**